libs/coin-modules/coin-celo/src/bridge/buildTransaction.ts

```diff
@@ -105,25 +113,145 @@ const buildTransaction = async (account: CeloAccount, transaction: Transaction)
+    //   .transfer(transaction.recipient, transaction.amount.toFixed())
+    //   .txo.encodeABI()
+    //   .slice(2);
+    // const contract = new ethers.utils.Interface(ERC20ABI);
+    // const data = contract
+    //   .encodeFunctionData("transfer", [transaction.recipient, transaction.amount.toFixed()])
+    //   .slice(2);
+
+    // const erc20Abi = [
+    //   {
+    //     constant: false,
+    //     inputs: [
+    //       { name: "_to", type: "address" },
+    //       { name: "_value", type: "uint256" },
+    //     ],
+    //     name: "transfer",
+    //     outputs: [{ name: "", type: "bool" }],
+    //     type: "function",
+    //   },
+    //   {
+    //     constant: true,
+    //     inputs: [],
+    //     name: "decimals",
+    //     outputs: [{ name: "", type: "uint8" }],
+    //     type: "function",
+    //   },
+    // ] as AbiItem[];
+
+    // const usdtContract = new kit.connection.web3.eth.Contract(erc20Abi, usdtAddress);
+    // const recipient = transaction.recipient;
+    // const decimals = await usdtContract.methods.decimals().call();
+    // let data;
+    // if (transaction.amount.gt(0)) {
+    //   const amount = kit.connection.web3.utils
+    //     .toBN(transaction.amount.toString())
+    //     .mul(kit.connection.web3.utils.toBN(10).pow(kit.connection.web3.utils.toBN(decimals)));
+    //   const transfer = usdtContract.methods.transfer(recipient, amount);
+    //   data = transfer.encodeABI();
+    // }
+
+    const client = createPublicClient({
+      transport: http(celo.rpcUrls.default.http[0] as string),
+      chain: celo,
+    });
+
+    const walletClient = createWalletClient({
+      transport: http(celo.rpcUrls.default.http[0] as string),
+      chain: celo,
+    });
+
+    const usdtAbi = [
+      {
+        constant: false,
+        inputs: [
+          { name: "_to", type: "address" },
+          { name: "_value", type: "uint256" },
+        ],
+        name: "transfer",
+        outputs: [{ name: "", type: "bool" }],
+        type: "function",
+      },
+    ];
+
+    // const usdtAddress = "0x48065fbBE25f71C9282ddf5e1cD6D6A887483D5e";
+
+    const data = await encodeFunctionData({
+      abi: usdtAbi,
+      functionName: "transfer",
+      args: [transaction.recipient, 1000000n],
+    }).slice(2);
+
+    const block = await client.getBlock({ blockTag: "latest" });
+    const baseFee = block.baseFeePerGas || BigInt(30);
+    const priorityFee = 1_000_000n;
+    const maxFeePerGas = baseFee + priorityFee;
+
+    const nonce = await client.getTransactionCount({
+      address: account.freshAddress as `0x${string}`,
+      blockTag: "latest",
+    });
 
     celoTransaction = {
       ...celoTransaction,
-      gas,
+      // data: token.transfer(transaction.recipient, value.toFixed()).txo.encodeABI(),
+      data,
+      gas: BigNumber(baseFee.toString()).multipliedBy(0.0000015).toNumber(),
+      maxFeePerGas: maxFeePerGas.toString(),
+      maxPriorityFeePerGas: priorityFee.toString(),
```

Comment:
sometimes  .toString() can return something that looks like scientific notation which breaks stuff